### PR TITLE
Update WP for latest BAP

### PR DESCRIPTION
**Pull request by fortunac**
This PR updates WP to work with BAP version 2.2.0-alpha+e8d63c0.

### CODE REVIEW

File: wp/lib/bap_wp/src/dune [hunk 1]
```diff
@@ -1,4 +1,4 @@
 (library
    (name bap_wp)
    (public_name bap_wp)
-   (libraries bap bap-x86-cpu z3 oUnit))
+   (libraries bap bap-x86-cpu z3 oUnit re))
```
Comment (jtpaasch) [inline wp/lib/bap_wp/src/dune:4]: Should we change this to ounit2?

And also modify the `.opam` file too?

File: wp/lib/bap_wp/src/symbol.ml [hunk 2]
```diff
@@ -26,7 +26,7 @@ let build_command (filename : string) : string =
   Format.sprintf "objdump -t %s | sed '0,/SYMBOL TABLE:/d' | grep -E 'data|bss' | awk '{print $1, $NF}'" filename
 
 let run_command (cmd : string) : string list =
-  let inp = Unix.open_process_in cmd in
+  let inp = Caml_unix.open_process_in cmd in
```
Comment (jtpaasch) [inline wp/lib/bap_wp/src/symbol.ml:29]: So we're moving to the standard library `Unix` module. :thumbsup: Me likes not having dependencies on other libraries for simple matters like this.

File: wp/lib/bap_wp/src/symbol.ml [hunk 3]
```diff
@@ -26,7 +26,7 @@ let build_command (filename : string) : string =
...
   let r = In_channel.input_lines inp in
   In_channel.close inp; r
```
Comment (jtpaasch) [inline wp/lib/bap_wp/src/symbol.ml:31]: If `open_process_in` throws an exception, you don't catch that, so you'll have a channel that's not properly closed when the program terminates. That isn't a big deal, unless you're spawning lots of processes, or you don't ever reboot your machine.

This isn't something to fix now, but maybe a ticket should be opened for it?

(Good examples of ensuring channels are closed can be found in the source code for the standard `Unix` module's `open_process*` functions, or in the `scram` module.)

File: wp/plugin/.merlin [hunk 4]
```diff
@@ -1,13 +1,11 @@
 S ./wp.ml
-S ./tests
-S ./lib
+S ./tests/**
+S ./lib/**
 
 PKG bap
 PKG core
 PKG z3
 PKG oUnit
```
Comment (jtpaasch) [inline wp/plugin/.merlin:8]: Change this to `ounit2` and include `re`?

File: wp/plugin/tests/test_libs/test_wp_utils.ml [hunk 5]
```diff
@@ -126,23 +126,22 @@ let check_list (expected_reg_models : ((string * string) list) list )
...
+                      reg value observed_value)
               end in
             Option.merge ~f:(^) acc err
           ) in
         begin match result with
           | None -> None
           | Some cur_err ->
             let model_msg = (Printf.sprintf "\nModel %d:\n" i) in
-            err ^ model_msg ^ cur_err |> Some
+            Some (err ^ model_msg ^ cur_err)
```
Comment (jtpaasch) [inline wp/plugin/tests/test_libs/test_wp_utils.ml:144]: :thumbsup: I like this better than `|> Some`

**Review by jtpaasch — APPROVED** [review]
I'm approving this, even though I've left a few questions about edits. I trust Chloe will make whatever changes she sees fit, and if she doesn't make any of the changes, she's got a good reason for it. I say fix what you want, and merge!

File: wp/lib/bap_wp/src/dune (hunk 1 above)
Comment (fortunac) [reply wp/lib/bap_wp/src/dune:4]: I can definitely change this!

**Comment by jtpaasch** [comment]
Merge it @fortunac

**Pull request merged**